python/cudf/io/csv.py

```diff
@@ -158,7 +158,7 @@ def read_csv(filepath_or_buffer, lineterminator='\n',
     # Populate csv_reader struct
     if is_file_like(filepath_or_buffer):
         if compression == 'infer':
-            raise ValueError("Cannot infer compression type from a buffer")
+            compression = None
```

Comment:
We should change this when we support decompression and scheme inference in the future, but for now I support this change.

Reply:
There are edge cases where people would want to send a compressed buffer as well.

Reply:
Sending a compressed buffer would still work, just not with 'infer' option. Since we use the file extension to infer the compression type, we cannot apply it to buffers.